test/e2e/getserversideprops/test/index.test.ts

```diff
@@ -599,7 +599,7 @@ const runTests = (isDev = false, isDeploy = false) => {
     )
     const $ = cheerio.load(html)
     const query = JSON.parse($('#query').text())
-    expect(query.post).toBe('post-1')
+    expect(query.post).toBe('something-else')
```

Comment:
Query string values should likely take precedence over params.

I think the original RFC was hinting at that `query` was supposed to be the query string (URL search params) only, anyway: https://github.com/vercel/next.js/issues/9524#:~:text=query%20%2D%20The%20query%20string%20(not%20entirely%20sure%20about%20this%20one%2C%20but%20probably%20needed)

Our docs says that `query` is:
> An object representing the query string, including dynamic route parameters.

But does not clarify if the query string value would be prioritized over the dynamic param value if they have the same key (ie. given `[id].js` and `/foo?id=bar`, should `query` resolve to `{id: "bar"}` instead of `{id: "foo"}`)? IMO it should. To get the dynamic param values, there is already `params`.

https://nextjs.org/docs/api-reference/data-fetching/get-server-side-props#context-parameter